### PR TITLE
refactor(intersect): update for Vue 3

### PR DESCRIPTION
**Pull request by sh7dm**
## Description
Refactor intersect directive for use with Vue 3.

## Motivation and Context
Upcoming Vuetify 3 release for Vue 3.

## How Has This Been Tested?
`jest`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
Requesting initial comments @vuetifyjs/core-contributors @vuetifyjs/core-team

File: packages/vuetify/src/directives/intersect/index.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { VNodeDirective } from 'vue/types/vnode'
+import { DirectiveBinding, ObjectDirective } from 'vue'
 
-interface ObserveVNodeDirective extends VNodeDirective {
+interface ObserveVNodeDirectiveBinding extends DirectiveBinding {
```
Comment (sh7dm) [inline packages/vuetify/src/directives/intersect/index.ts]: renaming:
```suggestion
interface ObserveDirectiveBinding extends DirectiveBinding {
```

File: packages/vuetify/src/directives/intersect/index.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
   options?: IntersectionObserverInit
 }
 
-function inserted (el: HTMLElement, binding: ObserveVNodeDirective) {
+function mounted (el: HTMLElement, binding: ObserveVNodeDirectiveBinding) {
```
Comment (sh7dm) [inline packages/vuetify/src/directives/intersect/index.ts]: renaming:
```suggestion
function mounted (el: HTMLElement, binding: ObserveDirectiveBinding) {
```

File: packages/vuetify/src/directives/intersect/__tests__/intersect.spec.ts [hunk 3]
```diff
@@ -1,38 +1,45 @@
 // Directives
 import Intersect from '../'
 
+// Libraries
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/directives/intersect/__tests__/intersect.spec.ts]: move before directive import

File: packages/vuetify/src/directives/intersect/index.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
-import { VNodeDirective } from 'vue/types/vnode'
+import { DirectiveBinding, ObjectDirective } from 'vue'
```
Comment (johnleider) [inline packages/vuetify/src/directives/intersect/index.ts]:
```suggestion
// Types
import {
  DirectiveBinding, 
  ObjectDirective,
} from 'vue'
```

File: packages/vuetify/src/directives/intersect/__tests__/intersect.spec.ts [hunk 5]
```diff
@@ -1,38 +1,45 @@
...
+import { h } from 'vue'
+
 // Directives
 import Intersect from '../'
 
 describe('resize.ts', () => {
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/directives/intersect/__tests__/intersect.spec.ts]:
```suggestion
describe('intersect.ts', () => {
```

File: packages/vuetify/src/directives/intersect/index.ts [hunk 6]
```diff
@@ -1,10 +1,14 @@
-import { VNodeDirective } from 'vue/types/vnode'
+// Types
+import {
+  DirectiveBinding, 
+  ObjectDirective,
+} from 'vue'
 
-interface ObserveVNodeDirective extends VNodeDirective {
+interface ObserveDirectiveBinding extends DirectiveBinding {
   options?: IntersectionObserverInit
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/directives/intersect/index.ts]: shouldn't it be 
```
  value: TypeOfHandler | {
    handler: TypeOfHandler
    options: IntersectionObserverInit
  }

```
?

File: packages/vuetify/src/directives/intersect/__tests__/intersect.spec.ts [hunk 7]
```diff
@@ -1,38 +1,45 @@
...
 
     expect((el as any)._observe).toBeTruthy()
     expect(callback).not.toHaveBeenCalled()
 
     document.body.removeChild(el)
 
-    Intersect.unbind(el)
+    Intersect.unmounted(el, {}, vnode, vnode)
 
     expect((el as any)._observe).toBeFalsy()
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/directives/intersect/__tests__/intersect.spec.ts]:
```suggestion
    expect((el as any)._observe).toBeUndefined()
```

**Review by KaelWD — APPROVED** [review]

**Pull request merged**